pkg/genericapiserver/options/server_run_options.go

```diff
@@ -213,12 +215,14 @@ func (s *ServerRunOptions) NewSelfClient(token string) (clientset.Interface, err
 		QPS:   50,
 		Burst: 100,
 	}
-	if s.SecurePort > 0 {
+	if s.SecurePort > 0 && len(s.TLSCAFile) > 0 {
```

Comment:
tested by generating certs as mentioned in https://coreos.com/kubernetes/docs/latest/openssl.html and patching `hack/local-up-cluster.sh` as follows

```
diff --git a/hack/local-up-cluster.sh b/hack/local-up-cluster.sh
index e580a3d..398deff 100755
--- a/hack/local-up-cluster.sh
+++ b/hack/local-up-cluster.sh
@@ -310,12 +310,16 @@ function start_apiserver {
       --service-cluster-ip-range="10.0.0.0/24" \
       --cloud-provider="${CLOUD_PROVIDER}" \
       --cloud-config="${CLOUD_CONFIG}" \
+      --tls-cert-file=/home/dims/k8s/apiserver.pem \
+      --tls-private-key-file=/home/dims/k8s/apiserver-key.pem \
+      --tls-ca-file=/home/dims/k8s/ca.pem \
+      --basic-auth-file=/home/dims/k8s/basicauth.csv \
       --cors-allowed-origins="${API_CORS_ALLOWED_ORIGINS}" >"${APISERVER_LOG}" 2>&1 &
     APISERVER_PID=$!

     # Wait for kube-apiserver to come up before launching the rest of the components.
     echo "Waiting for apiserver to come up"
-    kube::util::wait_for_url "http://${API_HOST}:${API_PORT}/api/v1/pods" "apiserver: " 1 ${WAIT_FOR_URL_API_SERVER} || exit 1
+    #kube::util::wait_for_url "http://${API_HOST}:${API_PORT}/api/v1/pods" "apiserver: " 1 ${WAIT_FOR_URL_API_SERVER} || exit 1
 }

 function start_controller_manager {
```